FeedStoreChallenge/CoreDataFeedCache.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  CoreDataFeedCache.swift
+//  FeedStoreChallenge
+//
+//  Created by Maciej Krolikowski on 08/04/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+//
+
+import Foundation
+import CoreData
+
+@objc(CoreDataFeedCache)
+public class CoreDataFeedCache: NSManagedObject {
+	@NSManaged public var timestamp: Date
+	@NSManaged public var feedItems: NSOrderedSet
+}
+
+extension CoreDataFeedCache {
+	static func deleteRequest() -> NSBatchDeleteRequest {
+		return NSBatchDeleteRequest(fetchRequest: CoreDataFeedCache.fetchRequest())
+	}
+
+	static func insert(feed: [LocalFeedImage], timestamp: Date, context: NSManagedObjectContext) {
+		let cache = self.init(context: context)
+		cache.feedItems = NSOrderedSet(array: feed.map { $0.toEntity(context: context) })
+		cache.timestamp = timestamp
+	}
+
+	static func fetch(context: NSManagedObjectContext) throws -> CoreDataFeedCache? {
+		let result = try context.fetch(CoreDataFeedCache.fetchRequest())
```

Comment:
You can remove `CoreDataFeedCache.` since they're in the same scope.

```suggestion
		let result = try context.fetch(fetchRequest())
```